### PR TITLE
opentelemetry-cpp: add versions 1.5.0 and 1.6.0

### PR DESCRIPTION
**Pull request by sproberts92**
Specify library name and version:  **opentelemetry-cpp/1.5.0** and **opentelemetry-cpp/1.6.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 1]
```diff
@@ -47,113 +65,112 @@ def requirements(self):
         self.requires("nlohmann_json/3.10.5")
         self.requires("openssl/1.1.1o")
```
Comment (uilianries) [inline recipes/opentelemetry-cpp/all/conanfile.py:66]:
```suggestion
        self.requires("openssl/1.1.1q")
```

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 2]
```diff
@@ -47,113 +65,112 @@ def requirements(self):
...
         self.requires("opentelemetry-proto/0.18.0")
-        self.requires("protobuf/3.21.1")
         self.requires("thrift/0.15.0")
-        if tools.Version(self.version) >= "1.3.0":
+        if Version(self.version) >= "1.3.0":
             self.requires("boost/1.79.0")
```
Comment (uilianries) [inline recipes/opentelemetry-cpp/all/conanfile.py:70]:
```suggestion
            self.requires("boost/1.80.0")
```

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 3]
```diff
@@ -47,113 +65,112 @@ def requirements(self):
         self.requires("nlohmann_json/3.10.5")
         self.requires("openssl/1.1.1o")
         self.requires("opentelemetry-proto/0.18.0")
-        self.requires("protobuf/3.21.1")
         self.requires("thrift/0.15.0")
```
Comment (uilianries) [inline recipes/opentelemetry-cpp/all/conanfile.py:68]:
```suggestion
        self.requires("thrift/0.17.0")
```

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 4]
```diff
@@ -47,113 +65,112 @@ def requirements(self):
         self.requires("nlohmann_json/3.10.5")
```
Comment (uilianries) [inline recipes/opentelemetry-cpp/all/conanfile.py:65]:
```suggestion
        self.requires("nlohmann_json/3.11.2")
```

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 5]
```diff
@@ -24,21 +29,34 @@ class OpenTelemetryCppConan(ConanFile):
...
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("abseil/20211102.0")
```
Comment (uilianries) [inline recipes/opentelemetry-cpp/all/conanfile.py:62]:
```suggestion
        self.requires("abseil/20220623.0")
```

**Review by uilianries — COMMENTED** [review]
Please, update dependencies versions to keep aligned with other recipes, otherwise, a conflict like this one will occur in the future again.

Also, need to update both `grpc/1.45.2` to 1.48.0 and `libcurl/7.83.1` to 7.85.0

You can obtain the latest version available on https://conan.io/center, or by running `conan search <package_name>  -r conancenter`

**Comment by sproberts92** [comment]
Thanks for the review @uilianries. I was away on holiday for the past month so I didn't get to look at this further until now. I see now #13299 has been merged already adding 1.6.1, so I will close this one.

**Pull request closed without merging**